app/javascript/components/upload/SequenceFileForm.js

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect } from 'react'
+
+import Select from 'lib/InstrumentedSelect'
+import FileUploadControl from './FileUploadControl'
+import { TextFormField, SavingOverlay, SaveDeleteButtons } from './form-components'
+
+/** renders a form for editing/uploading a sequence file and any assoicated bundle files */
+export default function SequenceFileForm({
+  file,
+  updateFile,
+  saveFile,
+  deleteFile,
+  addNewFile,
+  handleSaveResponse,
+  sequenceFileTypes,
+  fileMenuOptions,
+  associatedBaiFile
+}) {
+  const speciesOptions = fileMenuOptions.species.map(spec => ({ label: spec.common_name, value: spec.id }))
+  const selectedSpecies = speciesOptions.find(opt => opt.value === file.taxon_id)
```

Comment:
[The more I think about `taxon_id`](https://github.com/broadinstitute/single_cell_portal_core/pull/1181#discussion_r717723757), the more I think it's like `study_id` -- i.e., worth avoiding in client code.  It's used a lot for sequence files.

Like any study and `accession`, any file with an associated species (taxon) has a unique `ncbi_taxid` and `scientific_name`, e.g. "9606" and "Homo sapiens".  Unlike `taxon_id`, which is an opaque database ID tied to a particular MongoDB instance, both `ncbi_taxid` and (moreso) `scientific_name` are interoperable, memorable, and often known off-hand to users and developers.  

For better interoperability and often easier debugging, I suggest we use `ncbi_taxid` instead of database IDs as the canonical identifier for species throughout our API and clients.  I'd actually moreso prefer something like adding a `name` field with kebab-case formatting of `scientific_name` values.  Other bioinformatics APIs like Ensembl, NCBI, and MyGene.info all use scientific name or taxid as the canonical identifier in APIs.

This is relevant around now while we're doing a big refactor, have improved interoperability with moves from study accessions, and are considering more work in the CLI client.  I don't expect these changes in this PR, but figured here makes sense to outline a possible related improvement.

Reply:
makes sense to me.  I think the relative impact for this change with respect to our application codebase would be small (but it's also a relatively minor change--doesn't need a migration or anything like that).  And you're right on that it would make the CLI more comprehensible